js/apps/system/_admin/aardvark/APP/react/src/views/views/Components/ViewLinksBreadcrumbs.tsx

```diff
@@ -0,0 +1,110 @@
+import { Box, HStack, Text } from "@chakra-ui/react";
+import React from "react";
+import { useLinksContext } from "../LinksContext";
+
+
+const removeQuotes = (value: string) => {
+  return value?.split('"')[1];
+};
+
+const sanitize = (value: string) => {
+  return removeQuotes(value.split("]")[0]);
+};
+/**
+ * input: links["a"].fields["b"].fields["c"]
+ * output: [['links', 'a'], ['fields', 'b], ['fields', 'c']]
+ */
+const getFragments = (currentPath?: string) => {
+  if (!currentPath) {
+    return [];
+  }
+  // split by links[ and fields[
+  // const pathParts = currentPath.split(".");
+  // the above line doesn't work because field can contain dot
+  // so we need to split by .fields[ and .links[
+  const linkPathValue = removeQuotes(
+    currentPath.split("links[")[1].split("]")[0]
+  );
+  const fieldPathParts = currentPath.split(".fields[").slice(1);
+  // we need to append both the parts
+  const linkPathParts = ["links", linkPathValue];
+  let fragments = [linkPathParts]; // should be [['links', 'a'], ['fields', 'b], ['fields', 'c']]
+  fieldPathParts.forEach(pathPart => {
+    fragments = [...fragments, ["fields", sanitize(pathPart)]];
+  });
+  return fragments;
+};
+
+/**
+ * this function creates a path from the fragments
+ * input: [['links', 'a'], ['fields', 'b], ['fields', 'c']]
+ * output: links["a"].fields["b"].fields["c"]
+
+ *  */
+const getPathFromFragments = (fragments: string[][]) => {
+  let path = "";
+  fragments.forEach((fragment, index) => {
+    if (index === 0) {
+      path = fragment[0] + '["' + fragment[1] + '"]';
+    } else {
+      path = path + "." + fragment[0] + '["' + fragment[1] + '"]';
+    }
+  });
+  return path;
+};
+
+export const ViewLinksBreadcrumbs = () => {
+  const { setCurrentField, currentField } = useLinksContext();
```

Comment:
we set the `currentField` from three places
- clicking on a 'link'
- clicking on a 'field'
- clicking on an item in the breadcrumb

we use currentField.fieldPath to calculate fragments